mesonbuild/backend/backends.py

```diff
@@ -267,13 +299,17 @@ def get_compiler_options_for_target(self, target: build.BuildTarget) -> OptionOv
         comp_override = target.option_overrides_compiler
         return OptionOverrideProxy(comp_override, comp_reg)
 
-    def get_option_for_target(self, option_name: 'OptionKey', target: build.BuildTarget):
+    def get_option_for_target(self, option_name: 'OptionKey', target: build.BuildTarget) -> T.Union[str, int, bool, 'WrapMode']:
         if option_name in target.option_overrides_base:
             override = target.option_overrides_base[option_name]
-            return self.environment.coredata.validate_option_value(option_name, override)
-        return self.environment.coredata.get_option(option_name.evolve(subproject=target.subproject))
+            v = self.environment.coredata.validate_option_value(option_name, override)
+        else:
+            v = self.environment.coredata.get_option(option_name.evolve(subproject=target.subproject))
+        # We don't actually have wrapmode here to do an assert, so just do a
+        # cast, we know what's in coredata anyway.
+        return T.cast(T.Union[str, int, bool, 'WrapMode'], v)
```

Comment:
This should be more a more noticeable todo to remove this cast once we have annotated coredata.

Reply:
I don't think that we can properly annotate this, get_option returns  different type depending on the exact input, and I don't think there's a good way to annotate that, but I can add a better comment.